backend/module/bot/slackbot/slackbot.go

```diff
@@ -0,0 +1,249 @@
+		slack:         slack.New(config.BotToken),
+		signingSecret: config.SigningSecret,
+		bot:           bot,
+		logger:        logger,
+		scope:         scope,
+	}
+
+	return m, nil
+}
+
+type mod struct {
+	slack         *slack.Client
+	signingSecret string
+	bot           bot.Service
+	logger        *zap.Logger
+	scope         tally.Scope
+}
+
+func (m *mod) Register(r module.Registrar) error {
+	slackbotv1.RegisterSlackBotAPIServer(r.GRPCServer(), m)
+	return r.RegisterJSONGateway(slackbotv1.RegisterSlackBotAPIHandler)
+}
+
+// One request, one event. https://api.slack.com/apis/connections/events-api#the-events-api__receiving-events
+func (m *mod) Event(ctx context.Context, req *slackbotv1.EventRequest) (*slackbotv1.EventResponse, error) {
+	err := verifySlackRequest(ctx, req, m.signingSecret)
+	if err != nil {
+		m.logger.Error("verify slack request error", log.ErrorField(err))
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	m.logger.Info("received event from Slack API")
+
+	if req.Type == slackevents.URLVerification {
+		return m.handleURLVerificationEvent(req.Challenge)
+	}
+
+	// at this point in the flow, we can ack the Slack API with a 2xx response and process the event seperately
+	// TODO: (sperry) asynchronous processing
+	err = m.handleEvent(req)
+	if err != nil {
+		m.logger.Error("handle event error", log.ErrorField(err))
+	}
+
+	return &slackbotv1.EventResponse{}, nil
+}
+
+// retrieves the signature and timestamp from the headers which will be used to verify the request was from Slack
+func getSlackRequestHeaders(ctx context.Context) (http.Header, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, errors.New("no headers present on request")
+	}
+
+	signature := md.Get(hSignature)
+	if len(signature) == 0 {
+		return nil, fmt.Errorf("%s header not present on request", hSignature)
+	}
+
+	timestamp := md.Get(hTimestamp)
+	if len(timestamp) == 0 {
+		return nil, fmt.Errorf("%s header not present on request", hTimestamp)
+	}
+
+	// the slack package is case sensitive when looking up these headers
+	// https://github.com/slack-go/slack/blob/master/security.go#L56
+	xSignature := "X-Slack-Signature"
+	xTimestamp := "X-Slack-Request-Timestamp"
+
+	return http.Header{
+		xSignature: signature,
+		xTimestamp: timestamp,
+	}, nil
+}
+
+// verifies the request came from Slack
+// https://api.slack.com/authentication/verifying-requests-from-slack#verifying-requests-from-slack-using-signing-secrets__a-recipe-for-security__step-by-step-walk-through-for-validating-a-request
+func verifySlackRequest(ctx context.Context, req *slackbotv1.EventRequest, signingSecret string) error {
+	headers, err := getSlackRequestHeaders(ctx)
+	if err != nil {
+		return err
+	}
+
+	// returns a SecretsVerifier object in exchange for an http.Header object and signing secret
+	sv, err := slack.NewSecretsVerifier(headers, signingSecret)
+	if err != nil {
+		return err
+	}
+
+	body, err := protojson.Marshal(req)
```

Comment:
this is almost definitely not going to work. if the body is reordered or changed in anyway the signature won't match, and will definitely change after a roundtrip through the proto layer.

unfortunately, we're probably going to have to do some significant hacking to get at the raw input at this point in the code.

Reply:
just saw the screenshots... i'm perplexed as to how it would work.

here's an issue talking about access to raw request bodies https://github.com/grpc-ecosystem/grpc-gateway/issues/652

Reply:
sorry, i forgot to leave a PR todo that I'm still figuring out how to get this to work. locally I have this piece commented out

Reply:
ah thanks! wasn't exactly sure why this wasn't working. hmm okay going to brainstorm on this. we can also verify the request came from slack by checking the verification token sent in the request, though this flow in the PR is the preferred way.

Reply:
i'm going to come back to this in a follow up PR. Some things I've been thinking about
(1) change the API design to use `bytes` as the request body like in the example solution in the issue linked above
(2) verify request through mutual tls https://api.slack.com/authentication/verifying-requests-from-slack#verifying-requests-from-slack-using-signing-secrets__mutual-tls
(3) use mux to create a http handler for our slack events endpoint so that I can have access to the raw request body
(4) still make this work with the current API design but don't know what kind of hacking I need to do yet

for now in this PR, I'll verify the request came from Slack by checking the verification token sent by Slack in the request. Slack still supports this style of verification but this will be a short-term solution since signing secrets or a mutal tls is the more secure way.

Reply:
(2) mutual TLS is probably not viable because Clutch won't necessarily be deployed at the edge. if there's a proxy there (e.g. Envoy) doing the ability to verify from Clutch will be lost.

(3) too bespoke, circumvents middleware, etc. application logic should exist whenever possible in a gRPC handler.

Reply:
ty for the feedback! i'm going to try option #1 in the next PR.